pkg/aws/ec2/mock/mock.go

```diff
@@ -286,6 +287,47 @@ func (e *API) AssignPrivateIpAddresses(eniID string, addresses int64) error {
 	return fmt.Errorf("Unable to find ENI with ID %s", eniID)
 }
 
+func (e *API) UnassignPrivateIpAddresses(eniID string, addresses []string) error {
+	e.rateLimit()
+	e.simulateDelay(UnassignPrivateIpAddresses)
+
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
+	if err, ok := e.errors[UnassignPrivateIpAddresses]; ok {
+		return err
+	}
+
+	for _, enis := range e.enis {
+		if eni, ok := enis[eniID]; ok {
+			subnet, ok := e.subnets[eni.Subnet.ID]
+			if !ok {
+				return fmt.Errorf("subnet %s not found", eni.Subnet.ID)
+			}
+
+			releaseMap := make(map[string]int)
+			addressesAfterRelease := []string{}
+			for _, address := range addresses {
+				releaseMap[address] = 0
+			}
+
+			for _, address := range eni.Addresses {
+				_, ok := releaseMap[address]
+				if !ok {
+					addressesAfterRelease = append(addressesAfterRelease, address)
+				} else {
+					ip := net.ParseIP(address)
+					e.allocator.Release(ip)
+					subnet.AvailableAddresses++
+				}
+			}
+			eni.Addresses = addressesAfterRelease
+			return nil
```

Comment:
I don't understand this `return nil` here. You are iterating over `e.enis` why returning immediately after the finding one eni in the `enis` map?

Reply:
This function `UnassignPrivateIpAddresses` takes a given eniID and a list of addresses assigned to it.
e.enis is a map of {instanceID: [enis]}, so this is trying to iterate over instances and find the given eni, release the given addresses from it, and return immediately.